staker/staker.go

```diff
@@ -326,6 +333,9 @@ func NewStaker(
 			return nil, err
 		}
 	}
+	inactiveValidatedNodes := btree.NewG(2, func(a, b validatedNode) bool {
+		return a.number < b.number || (a.number == b.number && a.hash.Cmp(b.hash) < 0)
```

Comment:
Part of my concern here was when I used `Get`, it could return a validated node with a different hash than requested. I explicitly check for that too, but I thought it'd be safer if they were seen as different. That said, I think you're right that the most recently validated node will always be the canonical one, so we could change it.